pyproject.toml

```diff
@@ -1,17 +1,48 @@
-[tool.black]
-exclude = '''
-(
-  /(
-      \.eggs         # exclude a few common directories in the
-    | \.git          # root of the project
-  )/
-  | node_modules
-)
-'''
-
-[tool.isort]
-profile = "black"
-multi_line_output = 3
+[tool.ruff]
+ignore = ["E501"]
+exclude =[".git",".eggs","node_modules"] # exclude a few common directories in the root of the project
+
+[tool.ruff.lint]
+extend-select = [
+    "D",   # pydocstyle
+    "I",   # isort
+    "UP",  # pyupgrade
+    "ANN", # flake8-annotations
+]
+# Disable all "missing docstrings" and "missing type annotations" rules for now
+# TODO: enable
+ignore = [
+    "ANN001",
+    "ANN002",
+    "ANN003",
+    "ANN101",
+    "ANN102",
+    "ANN201",
+    "ANN202",
+    "ANN204",
+    "ANN205",
+    "ANN206",
+    "ANN401",
+    "D100",
+    "D101",
+    "D102",
+    "D103",
+    "D104",
+    "D105",
+    "D106",
+    "D200",
+    "D205",
+    "D400",
+    "D401",
+    "D404",
+    "UP007",
```

Comment:
I would already enable this one (you "just" need to replace `Union[...]` by `a | b` right ?

Reply:
Yes, that's right.